pyproject.toml

```diff
@@ -2,8 +2,8 @@
 # Minimum requirements for the build system to execute.
 # See https://github.com/scipy/scipy/pull/12940 for the AIX issue.
 requires = [
-    "meson-python>=0.13.1",
-    "meson>=1.2.1,<2",
+    "meson-python>=0.17.1",
+    "meson>=1.6.0,<2",
```

Comment:
This release is only a few months old - why can't older versions still be used?

Reply:
@WillAyd So we have two compilers in AIX. One is GCC supported from meson 1.3.0 and open XL compiler supported from 1.6.0. ([Read here](https://github.com/mesonbuild/meson/blob/master/docs/markdown/Release-notes-for-1.6.0.md#support-for-openxl-compiler-in-aix)) https://github.com/mesonbuild/meson/blob/master/docs/markdown/Release-notes-for-1.6.0.md#support-for-openxl-compiler-in-aix

So if Pandas uses meson >= 1.6.0 then in AIX, users can build Pandas using both the compilers. This is the sole reason we are using the latest version.

Reply:
You can use meson 1.6 on main today. This change is just making it so no one can use any other version

Reply:
What version would you like to change it to?

Reply:
Unless there is a feature we need generally from our build system I don't think this needs to be changed. There might be some simplifications we can get from upgrading to 1.4 with NumPy 2.0, but that should be done in tandem; a blanket bump of the version as shown in this PR doesn't provide a ton of value, unfortunately

Reply:
> So if Pandas uses meson >= 1.6.0 then in AIX, users can build Pandas using both the compilers. This is the sole reason we are using the latest version.

Just to reiterate what @WillAyd was saying, if pandas uses `meson >= 1.2.1`, users can still build pandas using both compilers. They just need to ensure they have meson 1.6.0 or higher installed.

Reply:
@rhshadrach @WillAyd Thank you for your comments. Yes, you are correct. I have modified the PR and squashed the commit. 

1.3.0 is okay for us :).

Reply:
Kindly let me know if this version is okay.

Reply:
You can currently build pandas with any version higher than 1.2.1. So without any changes, you can build pandas with 1.3.0 or 1.6.0.

Why does this need to change at all?

Reply:
@rhshadrach Kindly check the log here (https://github.com/pandas-dev/pandas/issues/60495#issuecomment-2526877435)

Meson 1.2.1 gets pulled to build, and AIX users complain they cannot build Pandas via pip. That is why we want to at least have 1.3.0 in pyproject.toml. To make sure a buggy meson is not used in any case to build pandas. Kindly consider 1.3.0 for the same.

Reply:
@KamathForAIX there is nothing that pins the version to 1.2.1; I think you just have that in the cache on your local system, so pip is happy to reuse it.

Does `pip3.11 install pandas -v --no-cache-dir` fix your issue?

Reply:
@WillAyd You are correct. Clearing the cache works.

pip3.11 install pandas -v --no-cache-dir, this works. I missed this. 

I agree with @rhshadrach. Thank you so much. It will be nice if you can merge this PR since users will not have the issue even if they have 1.2.1 in the cache. Ideally, folks do not clear their cache on their servers. 

Having said that, I can handle this in AIX forums internally. 

So whatever works for the community (be it merging this PR or closing this issue) is okay for me, Linux folks and everyone will do :)

Again, thank you so much. @rhshadrach @WillAyd 

Have a nice day.

Reply:
The problem is AIX is proprietary and such a small user base. It's not something that is officially supported, so its hard to justify making changes to pandas to accommodate it, when the majority of users might be totally fine with using meson 1.2

I think the workaround for AIX is the best solution for now

Reply:
@WillAyd - with 1.3 having been released November 2024, any thoughts on when we would feel good about bumping up to it as the minimum?

Reply:
I think the bump should occur when we feel like there is a feature that is worth taking advantage (for instance, I think 1.4 could simplify our configurations a lot since it handles numpy 2.0 more natively)

Outside of features though, I don't think there is any advantage to bumping. Build dependencies like meson have a pretty long shelf life

Reply:
> I don't think there is any advantage to bumping

Not sure how significant it is, but making it less error-prone to install pandas on even an uncommon platform seems like an advantage to me.

Reply:
Its not a hill I'm going to die on, but you are asking everyone currently developing or installing happily on meson 1.2 to upgrade their meson version for AIX users that have 1.2 in their pip cache